array_test.go

```diff
@@ -108,6 +108,61 @@ func TestArray_Constructors(t *testing.T) {
 	})
 }
 
+func TestArray_Decode(t *testing.T) {
+	t.Run("Decode with slice of interface", func(t *testing.T) {
+		target := []interface{}{}
+		testValue := []interface{}{"Foo", 123.0}
+		reporter := newMockReporter(t)
+		arr := NewArray(reporter, testValue)
+		arr.Decode(&target)
+		arr.Equal(target)
+	})
+	t.Run("Decode with slice of struct", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		type S struct {
+			Foo int `json:"foo"`
+		}
+		testValue := []interface{}{
+			map[string]interface{}{
+				"foo": 123,
+			},
+			map[string]interface{}{
+				"foo": 456,
+			},
+		}
+		arr := NewArray(reporter, testValue)
+		target := new([]S)
```

Comment:
@gavv is this fine or use the slice denotation as I did below? I presume we should use one of those to stay consistent. Which one tho?

Reply:
You mean whether to use new([]s) or []s{} ?

I guess there is no difference, just a matter of tatste, so up to you.

Reply:
Maybe this way it would be more readable:

var target []S
arr.Decode(&target)

This way the type of target is clear, as well it's clear that we're passing pointer.

Same for the first test:

var target []interface{}
arr.Decode(&target)

Btw we also need third test:

var target interface{}
arr.Decode(&target)